python/src/iceberg/types.py

```diff
@@ -48,27 +46,69 @@ def __str__(self):
 
     @property
     def is_primitive(self) -> bool:
-        return self._is_primitive
+        return isinstance(self, PrimitiveType)
+
 
+class PrimitiveType(Type):
+    """Base class for all Iceberg Primitive Types"""
+
+    _implemented = {}  # type: ignore
```

Comment:
What about `_instances`? I don't quite follow what you mean by the name `_implemented`.

Reply:
the `implemented` thinking was back from when I was using that factory pattern for generics.  `instances` does sound better actually. changed it